middleware.js

```diff
@@ -12,11 +13,43 @@ function webpackHotMiddleware(compiler, opts) {
   var eventStream = createEventStream(opts.heartbeat);
   var latestStats = null;
 
-  compiler.plugin("compile", function() {
-    latestStats = null;
-    if (opts.log) opts.log("webpack building...");
-    eventStream.publish({action: "building"});
+  var compilers = compiler.compilers || [compiler];
+  compilers.forEach(function(compiler, index) {
+    compiler.id = "__WEBPACK_COMPILER_" + index + "__";
+    compiler.plugin("compilation", function (compilation, params) {
+      var hotUpdateChunkTemplate = compilation.hotUpdateChunkTemplate;
+      if(!hotUpdateChunkTemplate) return;
+
+      compilation.mainTemplate.plugin("require-extensions", function(source) {
```

Comment:
Yeah, I steal it from `webpack/lib/HotModuleReplacementPlugin.js`, and handle `__webpack_compiler__` with the same way to process `__webpack_hash__`(which is used in `webpack-hot-middleware/process-update.js`) in it.

I will make a comment in it as soon.